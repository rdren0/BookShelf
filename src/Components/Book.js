import React, {Component} from 'react'

export class Book extends Component{
    constructor(){
        super()
        this.state={nothing: ''}
    }
    consoleLog = (book) => {
        window.localStorage.setItem(book.volumeInfo.authors,  book.id);

    }

    render(){
        let book = this.props.book;
        console.log(book)
        return (
            <div className="book">
                <h5>{ book.volumeInfo.title }</h5>
                {book.volumeInfo.imageLinks && <img src={book.volumeInfo.imageLinks.thumbnail} alt="test" />}
                <p>{book.volumeInfo.authors}</p>
                <h6 onClick={() => this.consoleLog(book)}>CLICK HERE</h6>
            </div>
        )
    }
}

export default Book;