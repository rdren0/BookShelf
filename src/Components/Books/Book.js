import React, { Component } from "react";
import { Link } from "react-router-dom";


export class Book extends Component {
  
  addFavorite = (book, value) => {
    this.props.alterFavorites(book,value);
  };

  render() {
    let book = this.props.book;
    return (
      <div className="book">
        <h5>{book.volumeInfo.title}</h5>
        <Link to={`/${book.id}`}>

        {book.volumeInfo.imageLinks && (
          <img src={book.volumeInfo.imageLinks.thumbnail} alt="test" />
        )}
        </Link>
        <p>{book.volumeInfo.authors}</p>

        <h6 onClick={() => this.addFavorite(this.props.book, "add")}>ADD</h6>
        <h6 onClick={() => this.addFavorite(this.props.book.id, "delete")}>DELETE</h6>
      </div>
    );
  }
}

export default Book;