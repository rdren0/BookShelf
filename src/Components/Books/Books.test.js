import React from "react";
import { shallow } from "enzyme";
import { Book } from "./Book";

describe("Book", () => {
  let wrapper;
  beforeEach(() => {
    let book = { volumeInfo: { title: "Great Gatsby" } };
    let mockStore = {}
    let mockFN = jest.fn()
    wrapper = shallow(<Book book={book} store={mockStore} alterFavorites={mockFN} />);
  })
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an add button when favorited is false', () => {
    wrapper.setState({ favorited: false });
    expect(wrapper).toMatchSnapshot();
  })
  it('should render a delete button when favorited is true', () => {
    wrapper.setState({ favorited: true });
    expect(wrapper).toMatchSnapshot();
  })

  it('should call addFavorite when button is pressed', () => {
    let MockFn = jest.spyOn(wrapper.instance(), "addFavorite");
    let button  = wrapper.find('.favorites')
    button.simulate('click')
    expect(MockFn).toHaveBeenCalled()
  })
});
