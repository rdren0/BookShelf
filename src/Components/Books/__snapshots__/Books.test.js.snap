// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Book should render a delete button when favorited is true 1`] = `ShallowWrapper {}`;

exports[`Book should render an add button when favorited is false 1`] = `ShallowWrapper {}`;
